### ISSUE TITLE
Provide a way to remove error from AsyncLoading

### ISSUE THREAD

**Issue by HugoHeneault** [rrousselGit/riverpod#2093]
**Is your feature request related to a problem? Please describe.**
I'd like to have AsyncNotifier with `AsyncValue.guard`'s error reseted when I call it again.

**Describe the solution you'd like**
I have a signin controller like this: 

```dart
import 'package:riverpod_annotation/riverpod_annotation.dart';

// freezed
import '../../models/user.dart';

part 'signin_controller.g.dart';

@riverpod
class SigninController extends _$SigninController {
  @override
  FutureOr<User?> build() {
    return null;
  }

  Future<User?> login(String email, String password) async {
    state = const AsyncLoading();
    state = await AsyncValue.guard(
        () => ref.read(userRepositoryProvider).login(email, password));

    return state.value;
  }
}

```

If `ref.read(userRepositoryProvider).login(email, password))` throws an error, my `SigninController`'s state has an `.hasError == true`, which is fine.

But if I call `SigninController.login()` again, `.error` remains during loading state.

I would like to have this `.hasError` set to false (error removed) when I set `state = const AsyncLoading()` even if previous `SigninController` had errored.

**Describe alternatives you've considered**
? No ideas...

**Additional context**
Code already given ⬆️

**Comment by AhmedLSayed9** [comment]
You can use `unwrapPrevious()`:
```
asyncValue.unwrapPrevious().hasError
```

**Comment by rrousselGit** [comment]
What you're asking for isn't a thing. Why do you need this? The control should be given to you at the widget level if you need it

**Comment by HugoHeneault** [comment]
Thank you so much for your quick answers. Appreciate it! 🙏 

> What you're asking for isn't a thing. Why do you need this? The control should be given to you at the widget level if you need it

When I press the "try again" button, I want to have the error gone. 

With @AhmedLSayed9's answer (thanks alot!) I got it working. 

I don't get why I have to `unwrapPrevious().hasError` in my widget as I thought as soon as I set my `state = const AsyncLoading();` the new state wouldn't include previous error. 

But it works this way, so we can close. :)
